dependencies/src/test/java/com/google/cloud/tools/opensource/dependencies/RepositoryUtilityTest.java

```diff
@@ -57,16 +64,31 @@ public void testReadBom_coordinates() throws ArtifactDescriptorException {
   }
 
   @Test
-  public void testReadBom_path() throws MavenRepositoryException {
+  public void testReadBom_path() throws MavenRepositoryException, ArtifactDescriptorException {
     Path pomFile = Paths.get("..", "boms", "cloud-oss-bom", "pom.xml");
     
-    Bom bom = RepositoryUtility.readBom(pomFile);
+    Bom currentBom = RepositoryUtility.readBom(pomFile);
+    Bom oldBom = RepositoryUtility.readBom("com.google.cloud:libraries-bom:2.5.0");
+    ImmutableList<Artifact> currentArtifacts = currentBom.getManagedDependencies();
+    ImmutableList<Artifact> oldArtifacts = oldBom.getManagedDependencies();
+    
+    String coordinates = currentBom.getCoordinates();
+    Truth.assertThat(coordinates).startsWith("com.google.cloud:libraries-bom:");
+    Truth.assertThat(coordinates).endsWith("-SNAPSHOT");
     
-    ImmutableList<Artifact> artifacts = bom.getManagedDependencies();
-    Assert.assertEquals(217, artifacts.size());
-    String coordinates = bom.getCoordinates();
-    Assert.assertTrue(coordinates.startsWith("com.google.cloud:libraries-bom:"));
-    Assert.assertTrue(coordinates.endsWith("-SNAPSHOT"));
+    // This is a characterization test to verify that the managed dependencies haven't changed.
+    // However sometimes this list does change. If so, we want to 
+    // output the specific difference so we can manually verify whether
+    // the changes make sense. When they do make sense, we update the test. 
+    if (currentArtifacts.size() != 217) {
+        Set<Artifact> current = Sets.newHashSet(currentArtifacts);
+        Set<Artifact> old = Sets.newHashSet(oldArtifacts);
+        SetView<Artifact> currentMinusOld = Sets.difference(current, old);
+        String added = Joiner.on(", ").join(currentMinusOld);
+        SetView<Artifact> oldMinusCurrent = Sets.difference(old, current);
+        String subtracted = Joiner.on(", ").join(oldMinusCurrent);
+        Assert.fail("Dependency tree changed.\n  Added " + added + "\n  Removed" + subtracted);
```

Comment:
Regardless of the conversation above, I don't see how this is any better than what you had before, which was one line, and shows missing and unexpected elements.

```
assertThat(currentArtifacts).containsExactlyElementsIn(oldArtifacts);
```

Reply:
As you pointed out, that wasn't guarded and couldn't be changed when we needed it to be.

Reply:
done